src/nonlin_svm_model.jl

```diff
@@ -0,0 +1,155 @@
+export tanh_train_model, tanh_test_model#, tan_nls_model
+
+function tan_data_train()
+  #load data
+  A, b = MLDatasets.MNIST.traindata()
+  ind = findall(x -> x == 0 || x == 1, b)
+  #reshape to matrix
+  A = reshape(A,size(A,1)*size(A,2), size(A,3))./255
+
+  #get 0s and 1s
+  b = b[ind]
+  b[b.==0] .= -1
+  A = convert(Array{Float64, 2}, A[:, ind])
+  p = randperm(length(b))[1:Int(floor(length(b)/4))]
+  b = b[p]
+  A = A[:, p]
+  return A, b
+end
+
+function tan_data_test()
+  A0, b0 = MLDatasets.MNIST.testdata()
+  ind = findall(x -> x == 0 || x == 1, b)
+  A0 = reshape(A0,size(A0,1)*size(A0,2), size(A0,3))./255
+
+  #get 0s and 1s
+  b0 = b0[ind]
+  b0[b0.==0] .= -1
+  A0 = convert(Array{Float64, 2}, A0[:, ind])
```

Comment:
Looks like this whole bit could be a separate function that you'll call here and also in `tan_data_train()`.

Reply:
I suppose this depends on the above answer as well. I can call it "generate data" or something.

Reply:
Yes.